### PR TITLE
Added sample implementation of Infrastructure

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Add sample implementation of Infrastructure

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/7113

is needed for https://github.com/codenvy/che-docs/pull/2

### CODE REVIEW

**Review by gazarenkov — APPROVED** [review]

**Review by ghost — APPROVED** [review]

File: src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInfaModule.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import com.google.inject.multibindings.MapBinder;
+import com.google.inject.name.Names;
+import org.eclipse.che.api.workspace.server.spi.RuntimeInfrastructure;
+import org.eclipse.che.api.workspace.server.spi.environment.InternalEnvironmentFactory;
+import org.eclipse.che.workspace.infrastructure.dummy.bootstrapper.MyBootstrapperFactory;
+import org.eclipse.che.workspace.infrastructure.dummy.env.MyInternalEnvironment;
+import org.eclipse.che.workspace.infrastructure.dummy.env.MyInternalEnvironmentFactory;
+
+/** @author Sergii Leshchenko */
+public class MyInfaModule extends AbstractModule {
```
Comment (gorkem) [inline src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInfaModule.java:24]: Typo MyInfaModule -> MyInfraModule

File: src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInternalRuntime.java [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+      doWaitServersRunning(machineName, machineConfig);
+    } catch (Exception e) {
+      sendMachineFailedEvent(machineName, e.getMessage());
+      throw e;
+    }
+  }
+
+  private void doStartMachine(String machineName, InternalMachineConfig machineConfig) {
+    //
```
Comment (gorkem) [inline src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInternalRuntime.java:110]: consider adding some comment, should not be empty  `//`

File: README.md [hunk 3]
```diff
@@ -1,2 +1,5 @@
-# che-infra-sample
-Che dummy SPI implementation
+# Che Infra Sample
+
+Sample that shows possible structure for implementations of Infrastructure.
+It is not ready to use and contains comments what should be implemented in infrastructure specific way.
+To include the implemented infrastructure into assembly it is enough to bind `MyInfaModule`.
```
Comment (gorkem) [inline README.md:5]: into assembly ->into an assembly
Typo MyInfaModule -> MyInfraModule

**Comment by sleshchenko** [comment]
@gorkem Thanks for review.
I'll create another PR for fixing your comments.

File: README.md (hunk 3 above)
Comment (sleshchenko) [reply README.md:5]: thx. fixed by https://github.com/codenvy/che-infra-sample/pull/2

File: src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInfaModule.java (hunk 1 above)
Comment (sleshchenko) [reply src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInfaModule.java:24]: thx. fixed by https://github.com/codenvy/che-infra-sample/pull/2

File: src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInternalRuntime.java (hunk 2 above)
Comment (sleshchenko) [reply src/main/java/org/eclipse/che/workspace/infrastructure/dummy/MyInternalRuntime.java:110]: +1
Added comment about machine starting in https://github.com/codenvy/che-infra-sample/pull/2

**Pull request merged**